client/src/main/java/org/apache/celeborn/client/ShuffleClient.java

```diff
@@ -63,7 +70,16 @@ public static void reset() {
     hadoopFs = null;
   }
 
-  protected ShuffleClient() {}
+  protected ShuffleClient() {
+    transportMessagesRunner =
+        ThreadUtils.newDaemonSingleThreadExecutor("client-transport-messages-runner");
+    runTransportMessagesStaticBlockerTask =
+        transportMessagesRunner.submit(
```

Comment:
why should register all extensions for every shuffleClient?

Reply:
@RexXiong, each process has only one shuffle client, therefore it's necessary to register extension for each shuffle client.